### PR TITLE
[TS] LPS-181899 Item Selector won't show all Sites grouped to the paginated page

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

Could you please review this pull request?

Motivation
=======
[GroupItemSelectorProviderImpl._filterGroups](https://github.com/liferay/liferay-portal/blob/master/modules/apps/item-selector/item-selector-impl/src/main/java/com/liferay/item/selector/internal/provider/GroupItemSelectorProviderImpl.java#L123) filters out the Site that the User has no permission to view. The problem is that the pagination won't align with the modified list and thus won't display all Sites the User still has access to.

Proposed Solution
========
If the number of filtered groups ends up being less than that of original the original results in a paginated section, pad it with groups from the subsequent sections. This logic has been outsourced to the new class [GroupItemSelectorGroupUtil](https://github.com/liferay-lima/liferay-portal/commit/eaeeae7099ca6d11920ed845f4be0df4377c6afa), as agreed upon in https://github.com/liferay-lima/liferay-portal/pull/4221#discussion_r1175253005.

Steps to Verify
========
Please see the linked LPS for the reproduction steps.
https://issues.liferay.com/browse/LPS-181899

Thank you and best regards,
István

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by Jeremy-Ch** [comment]
The failures are unrelated.

**Comment by Jeremy-Ch** [comment]
ci:test:relevant

File: modules/apps/item-selector/item-selector-impl/src/main/java/com/liferay/item/selector/internal/GroupItemSelectorGroupUtil.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		long companyId, long[] classNameIds, String keywords) {
+
+		return GroupServiceUtil.searchCount(
+			companyId, classNameIds, keywords,
+			LinkedHashMapBuilder.<String, Object>put(
+				"actionId", ActionKeys.VIEW
+			).put(
+				"site", Boolean.TRUE
+			).build());
+	}
```
Comment (adolfopa) [inline modules/apps/item-selector/item-selector-impl/src/main/java/com/liferay/item/selector/internal/GroupItemSelectorGroupUtil.java:71]: Can you use this method in the provider? This same logic is replicated there in the `getGroupsCount` method.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/item-selector/item-selector-impl/src/main/java/com/liferay/item/selector/internal/GroupItemSelectorGroupUtil.java (hunk 1 above)
Comment (IstvanD) [reply modules/apps/item-selector/item-selector-impl/src/main/java/com/liferay/item/selector/internal/GroupItemSelectorGroupUtil.java:71]: @adolfopa I force-pushed the update.

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**